DevTools shortcut on Preferences and Waiver windows

araujoarthur0 opened the pull request:
#### Related issue
#245 

#### Context / Background
- Briefly explain the purpose of the PR by providing a summary of the context

Adding DevTools shortcut on Preferences and Waiver windows.

#### What change is being introduced by this PR?
- How did you approach this problem?
- What changes did you make to achieve the goal?
- What are the indirect and direct consequences of the change?

There's a new listener waiting for CTRL+SHIFT+I to be pressed in the preferences and waiver windows, to toggle chrome dev tools on those windows.

#### How will this be tested?
- How will you verify whether your changes worked as expected once merged?

All working.

tupaschoal commented on src/preferences.js:
```diff
@@ -66,4 +67,12 @@ $(() => {
     });
+
+    window.addEventListener('keyup', (event) => {
+        if (event.ctrlKey && event.shiftKey && (event.keyCode == 73 || event.keyCode == 105)) { // 'i' or 'I'
+            BrowserWindow.getFocusedWindow().webContents.toggleDevTools();
+            event.preventDefault();
+            return false;
+        }
```
Can we extract a function? It's very hardcoded with the `keyCode`s and all, so it'd be great if both places used the same call

araujoarthur0 replied:
Added it to a new window-aux.js file

araujoarthur0 commented:
Added a new window-aux.js file with the devtools binding function.
Took the chance to also extract alert and confirm functions to this aux file, and fixed the error string for when waiver periods don't contain working days.

tupaschoal commented on js/window-aux.js:
```diff
@@ -15,6 +15,30 @@ function bindDevToolsShortcut(window) {
+ */
+function showDialog(options, successCallback) {
+    options['title'] = options['title'] || 'Time to Leave';
+    dialog.showMessageBox(BrowserWindow.getFocusedWindow(), options).then(successCallback);
```
Is `then` called only on success?

araujoarthur0 replied:
The first argument of then is the success callback: https://masteringjs.io/tutorials/fundamentals/then

tupaschoal replied:
Ok :)

tupaschoal commented on changelog.txt:
```diff
@@ -14,6 +14,7 @@
 - Fix: [#250] Silently ignoring waiver on non-working day or non-working day range
 - Enhancement: [#228] Improved performance of TTL - Now moving through the calendar is much faster
 - Enhancement: [#152] Adding a "Copy" option in the "About message", making it easier to copy information when opening an issue
+- Enhancement: [#245] DevTools shortcut on Preferences and Waiver windows
```
Can you squash this changelog commit with the first one? I think it also would be useful to say something like:
```suggestion
- Enhancement: [#245] DevTools shortcut (Ctrl+Shift+I) on Preferences and Waiver windows
```

araujoarthur0 replied:
You mean not having a separate commit for the changelog?
When accepting the merge request are you not able to squash them on demand? (real question, I've never accepted one hahaha)

tupaschoal replied:
Yeah, I mean, you're basically changing the changelog (duh ahah) because of the first commit, so it should go together with it.

We can either squash them all together or have them all separate via the GitHub interface. We can squash portions using the CLI

thamara replied:
I've updated the changelog directly in master after merging the PR by mistake. (sorry again)

thamara commented:
Sorry everyone, I got confused and merged the PR into master.

araujoarthur0 commented:
Don't worry hahaha
